packages/xmtp/src/plugin.ts

```diff
@@ -14,16 +16,82 @@ import type {
 	PluginContext,
 	XmtpClient,
 	XmtpConversation,
-	XmtpMessage
+	XmtpMessage,
+	XmtpSender,
+	XmtpSubjects
 } from "@hybrd/types"
 import { logger } from "@hybrd/utils"
 import { randomUUID } from "node:crypto"
 import { createXMTPClient, getDbPath } from "./client"
 import { ContentTypeReply, ContentTypeText, type Reply } from "./index"
+import { AddressResolver } from "./resolver/address-resolver"
+import { BasenameResolver } from "./resolver/basename-resolver"
+import { ENSResolver } from "./resolver/ens-resolver"
+import { extractSubjects } from "./lib/subjects"
+import { createPublicClient, http } from "viem"
+import { mainnet, base } from "viem/chains"
 
 // Re-export types from @hybrd/types for backward compatibility
 export type { Plugin }
 
+// Create public clients for resolver instances (reused across messages)
+const mainnetClient = createPublicClient({
+	chain: mainnet,
+	transport: http()
+})
+
+const baseClient = createPublicClient({
+	chain: base,
+	transport: http()
+})
+
+/**
+ * Resolve sender information from message
+ * @param message - The XMTP message
+ * @param client - XMTP client instance
+ * @returns XmtpSender object with address, inboxId, name, and optional basename
+ */
+async function resolveSender(
```

Comment:
Just use the Resolver router top level class. It exposes helper functions for each of the resolver. Don't use individual resolvers.